wrappers/c/zxing.cpp

```diff
@@ -0,0 +1,202 @@
+{
+	ZXing::DecodeHints * cpphints = new ZXing::DecodeHints();
+	return (zxing_DecodeHints) cpphints;
+}
+
+void zxing_DecodeHints_free (zxing_DecodeHints hints)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	delete cpphints;
+}
+
+void zxing_DecodeHints_setTryHarder (zxing_DecodeHints hints, bool tryHarder)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	cpphints->setTryHarder(tryHarder);
+}
+
+void zxing_DecodeHints_setTryDownscale (zxing_DecodeHints hints, bool tryDownscale)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	cpphints->setTryDownscale(tryDownscale);
+}
+
+void zxing_DecodeHints_setFormats (zxing_DecodeHints hints, zxing_BarcodeFormat formats)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	ZXing::BarcodeFormat cppformats = (ZXing::BarcodeFormat) formats;
+	cpphints->setFormats(cppformats);
+}
+
+void zxing_DecodeHints_setBinarizer (zxing_DecodeHints hints, zxing_Binarizer binarizer)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	ZXing::Binarizer cppbinarizer = (ZXing::Binarizer) binarizer;
+	cpphints->setBinarizer(cppbinarizer);
+}
+
+void zxing_DecodeHints_setEanAddOnSymbol (zxing_DecodeHints hints, zxing_EanAddOnSymbol eanAddOnSymbol)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	ZXing::EanAddOnSymbol cppeanAddOnSymbol = (ZXing::EanAddOnSymbol) eanAddOnSymbol;
+	cpphints->setEanAddOnSymbol(cppeanAddOnSymbol);
+}
+
+void zxing_DecodeHints_setTextMode (zxing_DecodeHints hints, zxing_TextMode textMode)
+{
+	ZXing::DecodeHints * cpphints = (ZXing::DecodeHints *) hints;
+	ZXing::TextMode cpptextMode = (ZXing::TextMode) textMode;
+	cpphints->setTextMode(cpptextMode);
+}
+
+/*
+ * ...
+ */
+
+/*
+ * ZXing/Result.h
+ */
+
+void zxing_Result_free (zxing_Result result)
+{
+    ZXing::Result * cppresult = (ZXing::Result *) result;
+    delete cppresult;
+}
+
+bool zxing_Result_isValid (zxing_Result result)
+{
+	ZXing::Result * cppresult = (ZXing::Result *) result;
+	return cppresult->isValid();
+}
+
+zxing_BarcodeFormat zxing_Result_format (zxing_Result result)
+{
+	ZXing::Result * cppresult = (ZXing::Result *) result;
+	return (zxing_BarcodeFormat) cppresult->format();
+}
+
+/*
+ * ...
+ */
+
+/*
+ * ZXing/ReadBarcode.h
+ */
+
+zxing_Result zxing_ReadBarcode (const zxing_ImageView iv, const zxing_DecodeHints hints)
+{
+	const ZXing::ImageView & cppiv = (const ZXing::ImageView &) iv;
+	// TODO
+	//zxing.cpp: In function ‘void* zxing_ReadBarcode(zxing_ImageView, zxing_DecodeHints)’:
+	//zxing.cpp:157:76: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
+	//  157 |         const ZXing::DecodeHints & cpphints = (const ZXing::DecodeHints &) hints;
+	const ZXing::DecodeHints & cpphints = (const ZXing::DecodeHints &) hints;
+	ZXing::Result cppresult = ZXing::ReadBarcode(cppiv, cpphints);
+    ZXing::Result * cppresultptr = new ZXing::Result(cppresult);
```

Comment:
`new ZXing::Result(std::move(cppresult));` should be more performant and save a bunch of string copies.